frontend/editor/index.html

```diff
@@ -7,78 +7,94 @@
-		<option value="Break">Break</option>
-		<option value="AMSupport">AM Support</option>
-		<option value="Passing">Passing (implicit)</option>
-		<option value="BeforeSchool">BeforeSchool (implicit)</option>
-		<option value="AfterSchool">AfterSchool (implicit)</option>
-	</select>
-	<br>
-	<label for="period_class_index">Class Index</label>
-	<input id="period_class_index" type="number">
-	<script src="../editor/controls-section.js"></script>
+	<div id="settings">
+		<button id="pull">Fetch Schedules</button>
+		<button id="push">Apply Changes</button>
+		<button id="update">Force Schedule Reload</button>
+		<span id="controls_error" class="error"></span>
+		<hr>
+		<hr>
+		<h2>Manage Schedules</h2>
+		<button id="add_schedule">Create New Blank Schedule</button>
+		<br>
+		<select id="select_schedule">
+		</select>
+		<button id="copy_schedule">Duplicate Selected Schedule</button>
+		<button id="remove_schedule">Delete Selected Schedule</button>
+		<br>
+		<br>
+		<label for="schedule_friendly_name">Schedule Name</label>
+		<input id="schedule_friendly_name" class="schedule_setting">
+		<br>
+		<label for="schedule_code">Schedule Code (internal use)</label>
+		<input id="schedule_code" class="schedule_setting">
+		<br>
+		<label for="schedule_color">Schedule Color</label>
+		<input id="schedule_color" class="schedule_setting" type="color">
+		<br>
+		<label for="schedule_hide">Hide in dropdowns</label>
+		<input id="schedule_hide" class="schedule_setting" type="checkbox">
+		<br>
+		<label for="schedule_regex">Schedule Regex</label>
+		<input id="schedule_regex" class="schedule_setting">
+		<br>
+		<hr>
+		<label for="calendars">Calendar URLs (ical feed, one per line. Top items take priority.)</label>
+		<br>
+		<textarea id="calendars"></textarea>
+		<hr>
+		<h2>Manage Periods</h2>
+		<template id="period_settings">
+			<button id="move_up">Move Up</button>
+			<button id="move_down">Move Down</button>
+			<button id="remove_period">Remove</button>
+			<br>
+			<label for="period_friendly_name">Friendly Name</label>
+			<input id="period_friendly_name">
+			<br>
+			<label for="period_start">Start Time</label>
+			<input id="period_start">
+			<br>
+			<label for="period_end">End Time</label>
+			<input id="period_end">
+			<br>
+			<label for="period_kind">Period Kind</label>
+			<select id="period_kind">
+				<option value="Class">Class</option>
+				<option value="Lunch">Lunch</option>
+				<option value="Break">Break</option>
+				<option value="AMSupport">AM Support</option>
+				<option value="Passing">Passing (implicit)</option>
+				<option value="BeforeSchool">BeforeSchool (implicit)</option>
+				<option value="AfterSchool">AfterSchool (implicit)</option>
+			</select>
+			<br>
+			<label for="period_class_index">Class Code</label>
+			<input id="period_class_index">
+		</template>
+		<div id="periods">
+		</div>
+		<button id="add_period" disabled>Add Period</button>
+	</div>
 </body>
 
 </html>
```

Comment:
* Class names aren't saved at all.
* It's impossible to change a non-Class period into a Class.
* The Class ID field remains enabled when changing a period into a non-Class.
* The Class ID field remains disabled when changing a period into a Class.
* When creating a new schedule, the input to the prompt fills both the friendly_name and name fields, but when copying a schedule the friendly_name field is always `Copy of {previous}`.
* The schedule code should not be easy to modify, since it is intended as an immutable identifier. The class code should also present a confirmation message when edited.
* In some of my testing. the schedule editor caused HTTP 500 on the server, but I wasn't able to consistently replicate it.

Reply:
I could not reproduce the class name not saving issue. Besides that, all other issues (besides the 500 error) are fixed.